src/egraph.rs

```diff
@@ -428,7 +445,13 @@ impl<L: Language, N: Analysis<L>> EGraph<L, N> {
     /// Note that this function can be called again to explain any intermediate terms
     /// used in the output [`Explanation`].
     pub fn explain_existance(&mut self, expr: &RecExpr<L>) -> Explanation<L> {
-        let id = self.add_expr_internal(expr);
+        let id = self.add_expr_uncanonical(expr);
+        self.explain_existance_id(id)
+    }
+
+    /// Equivalent to calling [`explain_existance`](EGraph::explain_existance)`(`[`id_to_expr`](EGraph::id_to_expr)`(id))`
+    /// but more efficient
```

Comment:
Good documentation, but I think it needs a little more explanation.
Perhaps something like a warning that you need uncanonical ids, and how you might get those uncanonical ids in the first place.